config/7.3.0/fapi1part2adv/ig/scripts/groovy/ErrorResponseFilter.groovy

```diff
@@ -0,0 +1,44 @@
+/**
+ * This script is a simple implementation of HTTP Basic Authentication on
+ * server side.
+ * It expects the following arguments:
+ *  - realm: the realm to display when the user-agent prompts for
+ *    username and password if none were provided.
+ *  - username: the expected username
+ *  - password: the expected password
+ */
+SCRIPT_NAME = "[ErrorResponseFilter] - "
+
+logger.debug(SCRIPT_NAME + "Running...")
+
+class ErrorResponse {
+    public String error_description
+    public String error
+}
+
+return next.handle(context, request).thenOnResult {
+    response -> {
+        if(response.status.isClientError()){
+            logger.debug("{} Response has status {} - which is a Client Error", SCRIPT_NAME, response.status)
+            var errorResponse = response.entity.getJson();
+
+            logger.debug("{} Response has body {}", SCRIPT_NAME, errorResponse)
+            try {
+                ErrorResponse error = (ErrorResponse)errorResponse
```

Comment:
Does this work?

The getJson method will decode the byte[] received from the socket and produce a LinkedHashMap<String, Object> representation of the JSON. So it is my expectation that this cast will always fail.

Unfortunately we can't write a unit test for the groovy...

Reply:
That does work!

Reply:
As it's a map then you can use the dot notation in groovy to access keys in the map, so you can do:
```
if (errorResponse.error_description.equals("blah)
```

So the ErrorResponse class is not really needed.

Reply:
This works fine in bohocode-core

Reply:
Strange one, I wrote some java to do the same thing and it fails so must be some groovy magic. I'm kinda intrigued but regardless I think it is better with the change that you've made.